eyeshade/workers/reports.js

```diff
@@ -1356,49 +1372,52 @@ exports.workers = {
       if (publisher) {
         query = { publisher: publisher }
         entries = await settlements.find(query)
-        publishers = await mixer(debug, runtime, [ publisher ], range)
+        contributions = await mixer(debug, runtime, [ publisher ], range)
       } else {
         query = underscore.pick(payload, [ 'owner', 'hash', 'settlementId' ])
+        if (underscore.size(query) === 0) rollupP = false
         if (range) query._id = range
         entries = await settlements.find(query)
         if ((rollupP) && (entries.length > 0)) {
           query = { $or: [] }
           entries.forEach((entry) => { query.$or.push({ publisher: entry.publisher }) })
           entries = await settlements.find(query)
         }
-        if (payload.owner) {
-          publishers = await publishersC.find({ owner: payload.owner })
+        if (owner) {
+          publishers = await publishersC.find({ owner: owner })
           contributions = await mixer(debug, runtime, publishers.map((p) => p.publisher), range)
         } else {
           contributions = await mixer(debug, runtime, undefined, range)
         }
       }
 
-      referralTotals = await referrals.aggregate([
-        { $match: { owner: payload.owner } },
-        {
-          $group: {
-            _id: '$publisher',
-            count: { $sum: 1 },
-            probi: { $sum: '$probi' }
+      if (owner) {
+        referralTotals = await referrals.aggregate([
+          { $match: { owner: owner } },
+          {
+            $group: {
+              _id: '$publisher',
+              count: { $sum: 1 },
+              probi: { $sum: '$probi' }
+            }
           }
-        }
-      ])
-      referralTotals = referralTotals.reduce((obj, item) => {
-        item.publisher = item._id
-        item.probi = new BigNumber(item.probi.toString())
-        item.fees = item.probi.times(feePercent).truncated()
-        item.probi = item.probi.minus(item.fees)
-        item.altcurrency = altcurrency
-        item.note = 'Finalized referrals'
-
-        obj[item._id] = item
-        return obj
-      }, {})
+        ])
+        referralTotals = referralTotals.reduce((obj, item) => {
+          item.publisher = item._id
+          item.probi = new BigNumber(item.probi.toString())
+          item.fees = item.probi.times(feePercent).truncated()
+          item.probi = item.probi.minus(item.fees)
+          item.altcurrency = altcurrency
+          item.note = 'Finalized referrals'
+
+          obj[item._id] = item
+          return obj
+        }, {})
+      } else referralTotals = {}
 
       publishers = underscore.keys(contributions)
-      if (payload.owner) {
-        publishers.push(...underscore.keys(referralTotals))
+      if (owner) {
+        publishers.push(underscore.keys(referralTotals))
```

Comment:
I think we want the spread operator that was removed here. We are grouping by publisher above and can potentially receive multiple elements back.